core/src/impl/Kokkos_Core.cpp

```diff
@@ -167,20 +167,43 @@ bool is_valid_map_device_id_by(std::string const& x) {
 
 }  // namespace
 
+std::vector<int> const& Kokkos::Impl::get_visible_devices() {
+  static auto devices = get_visible_devices(get_device_count());
+  return devices;
+}
```

Comment:
Any objections to specialize for `SYCL` (in a folow-up) to store sycl::devices in that case? I am just now looking at implementing https://kokkosteam.slack.com/archives/C5BGU5NDQ/p1705683106903069 for filtering suitable `SYCL` devices.

Reply:
How do you want to specialize?
Also I did not figure how to you use your slack link.

Reply:
See https://github.com/kokkos/kokkos/pull/6758.

Reply:
Nevermind you do `get_sycl_devices`